sdks/universal-router-sdk/src/swapRouter.ts

```diff
@@ -82,14 +83,30 @@ export abstract class SwapRouter {
     positionManagerOverride?: string
   ): MethodParameters {
     const v4Pool: V4Pool = options.outputPosition.pool
-    const token0 = options.inputPosition.pool.token0
-    const token1 = options.inputPosition.pool.token1
+    let token0: Currency = options.inputPosition.pool.token0
+    let token1: Currency = options.inputPosition.pool.token1
     const v4PositionManagerAddress =
       positionManagerOverride ?? CHAIN_TO_ADDRESSES_MAP[v4Pool.chainId as SupportedChainsType].v4PositionManagerAddress
 
+    const nativeCurrency = Ether.onChain(v4Pool.chainId)
+
+    // if migrating wrapped native -> native
+    if (nativeCurrency.wrapped.equals(token0) || nativeCurrency.wrapped.equals(token1)) {
+      // v4 pool must use native currency and must have useNative flag set
+      invariant(v4Pool.token0.isNative === options.v4AddLiquidityOptions.useNative, 'NATIVE_REQUIRED')
```

Comment:
the comment above this doesnt align with this check? shouldnt it use `&` not `===`?